pyfans/micro.cpp

```diff
@@ -0,0 +1,118 @@
+// Micro simulation for mechanical problems
+// In this file we solve a micro problem with FANS which is controlled by the Micro Manager
+// This file is compiled with pybind11 to be available as a python module
+//
+// To check if python is able to import it, run:
+// python3 -c "import micro; micro.MicroSimulation(1)"
+// from the same directory
+
+#include "micro.hpp"
+#include "setup.h"
+#include "matmodel.h"
+
+py::array_t<double> merge_arrays(py::array_t<double> array1, py::array_t<double> array2)
+{
+    // Ensure arrays are contiguous for efficient merging
+    array1 = array1.attr("copy")();
+    array2 = array2.attr("copy")();
+
+    // Get numpy concatenate function
+    py::object np          = py::module::import("numpy");
+    py::object concatenate = np.attr("concatenate");
+
+    // Concatenate the two arrays
+    py::tuple           arrays = py::make_tuple(array1, array2);
+    py::array_t<double> result = concatenate(arrays, py::int_(0)).cast<py::array_t<double>>();
+
+    return result;
+}
+
+// Constructor
+MicroSimulation::MicroSimulation(int sim_id)
+{
+    MPI_Init(NULL, NULL);
+
+    // initialize fftw mpi
+    fftw_mpi_init();
+
+    // Input file name is hardcoded. TODO: Make it configurable
+    const char *input_files_path        = "input.json";
```

Comment:
If it is easy to pass the path as a parameter, I would find that sensible

Reply:
This is sadly not possible right now. The MicroSimulation class is imported by the Micro Manager, and then objects of it are created. So there is no parameter passing as such. In the future we can explore passing the `input.json` file as a constructor parameter of this class.